src/main/java/com/onarandombox/MultiverseCore/commands/CheckCommand.java

```diff
@@ -35,6 +35,11 @@ public void onCheckCommand(BukkitCommandIssuer issuer,
     ) {
         issuer.sendMessage("Checking " + player + " to " + destination + "...");
         //TODO More detailed output on permissions required.
-        this.plugin.getDestinationsProvider().checkTeleportPermissions(issuer, player, destination);
+        if (this.plugin.getPlayerActionChecker().canUseDestinationToTeleport(issuer.getIssuer(), player, destination).isUnsuccessful()) {
+            issuer.sendMessage("You don't have permission to use this destination.");
+        }
+        if (this.plugin.getPlayerActionChecker().canGoToDestination(issuer.getIssuer(), player, destination).isUnsuccessful()) {
+            issuer.sendMessage("You don't have permission to teleport to this destination.");
+        }
```

Comment:
Looking at this again I feel like readability is a bit tough. Can we shorten these if statements to just `if (canUseDestinationToTeleport(...)) {`

Reply:
Musing about this again, sorry... I feel that just in general `canUseDestinationToTeleport` feels like it should return a boolean. When I see `canUseDestinationToTeleport.isUnsuccessful` or `canUseDestinationToTeleport.isSuccessful` it strikes me as strange.

Reply:
I remove isUnsuccessful and isSuccessful, and changed them to just asBoolean.